hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -2712,16 +2714,52 @@ public ClusterMetrics getClusterMetricsWithoutCoprocessor(EnumSet<Option> option
       options = EnumSet.allOf(Option.class);
     }
 
+    // TASKS and/or LIVE_SERVERS will populate this map, which will be given to the builder if
+    // not null after option processing completes.
+    Map<ServerName, ServerMetrics> serverMetricsMap = null;
+    boolean processedLiveServers = false;
+
     for (Option opt : options) {
       switch (opt) {
         case HBASE_VERSION: builder.setHBaseVersion(VersionInfo.getVersion()); break;
         case CLUSTER_ID: builder.setClusterId(getClusterId()); break;
         case MASTER: builder.setMasterName(getServerName()); break;
         case BACKUP_MASTERS: builder.setBackerMasterNames(getBackupMasters()); break;
+        case TASKS: {
+          // Master tasks
+          builder.setMasterTasks(TaskMonitor.get().getTasks().stream()
+            .map(task -> ServerTaskBuilder.newBuilder()
+              .setDescription(task.getDescription())
+              .setStatus(task.getStatus())
+              .setState(ServerTask.State.valueOf(task.getState().name()))
+              .setStartTime(task.getStartTime())
+              .setCompletionTime(task.getCompletionTimestamp())
+              .build())
+            .collect(Collectors.toList()));
+          // TASKS is also synonymous with LIVE_SERVERS for now because task information for
+          // regionservers is carried in ServerLoad.
+          // Add entries to serverMetricsMap for all live servers
+          if (serverManager != null && !processedLiveServers) {
+            if (serverMetricsMap == null) {
+              serverMetricsMap = new HashMap<>();
+            }
+            final Map<ServerName, ServerMetrics> map = serverMetricsMap;
+            serverManager.getOnlineServers().entrySet()
```

Comment:
nit: could we factor out the common code into a helper rather than duplicating?

Reply:
I did take up your suggestion to D.R.Y this and was able to remove some lines, it looks good @gjacoby126